site/src/pages/components/enhanced-range-input.explainer.mdx

```diff
@@ -0,0 +1,475 @@
+```
+
+This fragmentation makes it difficult for developers to create consistent range input styles across browsers, often requiring browser-specific code or fallbacks.
+
+## Proposed Solution
+
+### Standardized Anatomy
+
+We propose standardizing the internal structure of range inputs across all browsers. This will ensure consistent styling and behavior, regardless of the browser implementation.
+
+<RangeAnatomy/>
+
+The standardized anatomy includes:
+
+- `::range-track`: The main track along which the thumb(s) move.
+- `::range-fill`: The filled portion of the track, typically between the minimum value and the thumb (or between thumbs for multi-handle ranges).
+- `::range-thumb`: The draggable handle(s) used to select values.
+- `::range-segment`: Sections of the track between handles in multi-handle ranges.
+- `::range-tick`: Optional tick marks along the track for value representation. (when datalist is paired).
+- `::range-tick-label`: Labels associated with tick marks. (when datalist is paired).
+
+### Styling API
+
+To enable comprehensive styling of range inputs, we propose introducing a new CSS `appearance` value and a set of pseudo-elements:
+
+```css
+input[type="range"] {
+    appearance: base-range;
+}
+```
+
+This opt-in approach allows developers to access enhanced styling capabilities while maintaining backwards compatibility.
+
+### Multi-Handle Support
+
+We propose extending the range input to support multiple handles, starting with dual-handle support as the most common use case. This will be achieved through new HTML attributes and corresponding JavaScript APIs.
+
+### Accessibility Enhancements
+
+To improve accessibility, especially for multi-handle ranges, we propose making individual thumbs focusable and providing clear audio cues for screen readers.
+
+### Backwards Compatibility
+
+The reason for opting in is to ensure backward compatibility with existing range inputs. Since current implementations are a mess, it feels like a safer bet. Browsers that do not support the new features will fall back to the current behavior of range inputs, and it opens up the window of targeting support via CSS (@supports).
+
+For multi-handle ranges, browsers that do not support the feature will treat the input as a standard single-handle range, using the first value provided in the `value` attribute
+
+### Datalist Integration
+
+We propose standardizing and enhancing the integration of `<datalist>` with range inputs. This will allow for consistent implementation of tick marks and predefined values across browsers, especially when used with the `appearance: base-range` styling.
+
+Key features:
+
+1. Standardized rendering of `<datalist>` options as tick marks on the range track.
+2. Ability to style tick marks using the `::range-tick` pseudo-element.
+3. Ability to style labels of tick marks using the `::range-tick-label` pseudo-element.
+4. Support for `<datalist>` with multi-handle range inputs.
```

Comment:
How? Just values in the options like the `value` syntax? E.g. `<option>23.2  34.5</option>`?

Reply:
I think the ticks should actually remain the same in that case no? it should just be able to move 2 handles alongside of them